src/main/java/frc/robot/subsystems/TrapArmSubsystem.java

```diff
@@ -0,0 +1,224 @@
+    groundBreak = new DigitalInput(TrapArmConstants.groundBreak_ID);
+
+    // Arm
+    baseMotor1 = new CANSparkMax(TrapArmConstants.baseMotor1_ID, MotorType.kBrushless);
+    baseMotor1.restoreFactoryDefaults();
+    baseMotor1.getPIDController().setP(basePID[0]);
+    baseMotor1.getPIDController().setI(basePID[1]);
+    baseMotor1.getPIDController().setD(basePID[2]);
+    baseMotor1.getPIDController().setIZone(basePID[3]);
+    baseMotor1.getPIDController().setFF(basePID[4]);
+
+    baseMotor2 = new CANSparkMax(TrapArmConstants.baseMotor2_ID, MotorType.kBrushless);
+    baseMotor2.restoreFactoryDefaults();
+    baseMotor2.getPIDController().setP(basePID[0]);
+    baseMotor2.getPIDController().setI(basePID[1]);
+    baseMotor2.getPIDController().setD(basePID[2]);
+    baseMotor2.getPIDController().setIZone(basePID[3]);
+    baseMotor2.getPIDController().setFF(basePID[4]);
+
+    baseBreak = new DigitalInput(TrapArmConstants.baseBreak_ID);
+
+    scoringMotor = new CANSparkMax(TrapArmConstants.scoringMotor_ID, MotorType.kBrushless);
+    scoringMotor.restoreFactoryDefaults();
+    scoringMotor.getPIDController().setP(scoringPID[0]);
+    scoringMotor.getPIDController().setI(scoringPID[1]);
+    scoringMotor.getPIDController().setD(scoringPID[2]);
+    scoringMotor.getPIDController().setIZone(scoringPID[3]);
+    scoringMotor.getPIDController().setFF(scoringPID[4]);
+
+    scoringBreak = new DigitalInput(TrapArmConstants.scoringBreak_ID);
+
+    // SmartDashboard
+    SmartDashboard.putBoolean("Arm Zeroed", (baseBreak.get() && scoringBreak.get()));
+  }
+
+  // Commands
+  public Command intakeSource() {
+    return startEnd(
+        () -> {
+          if (!sourceBreak.get()) {
+            setTrapArm(TrapArmState.FROM_SOURCE);
+            rollerMotor.set(TrapArmConstants.rollerIntakeSpeed);
+          }
+        },
+        () -> {
+          rollerMotor.stopMotor();
+        });
+  }
+
+  public Command intakeGround() {
+    return startEnd(
+        () -> {
+          if (!groundBreak.get()) {
+            setTrapArm(TrapArmState.FROM_INTAKE);
+            rollerMotor.set(-TrapArmConstants.rollerIntakeSpeed);
+          }
+        },
+        () -> {
+          rollerMotor.stopMotor();
+        });
+  }
+
+  public Command scoreAMP() {
+    return startEnd(
+        () -> {
+          setTrapArm(TrapArmState.AMP_SCORE);
+          rollerMotor.set(-TrapArmConstants.rollerScoringSpeed);
+        },
+        () -> {
+          rollerMotor.stopMotor();
+        });
+  }
+
+  public Command scoreTrap() {
+    return startEnd(
+        () -> {
+          setTrapArm(TrapArmState.TRAP_SCORE);
+          rollerMotor.set(TrapArmConstants.rollerScoringSpeed);
+        },
+        () -> {
+          rollerMotor.stopMotor();
+        });
+  }
+
+  public Command zeroArm() {
+    return startEnd(
+      () -> {
+        if (!baseBreak.get()) {
+          baseMotor1.getPIDController().setReference(10, ControlType.kVelocity);
```

Comment:
Should we have a constant for this?  ( Assume this is move slowly. )

Reply:
done